cmake/third_party.cmake

```diff
@@ -17,6 +17,7 @@ Include(ExternalProject)
 Include(FetchContent)
 
 option (WITH_UNWIND "Enable libunwind support" ON)
+option (LEGACY_GLOG "whether to use legacy glog library" ON)
```

Comment:
I tried to play around with variable visibility but I couldn't find a better way than duplicating the option. If there is a clever way to propagate this from the code that includes this I am all eyes :eyes:

Reply:
can you explain what the problem is?

Reply:
you can also remove `LEGACY_GLOG` from `CMakeLists.txt` and leave it only here

Reply:
Yes, option `LEGACY_GLOG` is not seen by `third_party.cmake`. You define the variable/option in `CmakeLists` which includes `third_party.cmake`. The problem is that the code included, does not observe the variable `LEGACY_GLOG` so it doesn't fetch anytthing

Reply:
> you can also remove LEGACY_GLOG from CMakeLists.txt and leave it only here 

Sounds good

Reply:
it is seen if you build helio separately. it is not seen in dragonfly. keeping it only here might fix the issue.

Reply:
yes otherwise helio builds would have failed -- it makes sense. What `I wanted` to understand is why the variables are not visible. I did google it but I couldn't find something relevant and I also tried to use different scopes and replacing `OPTION` with `SET`. None of my attempts worked so I ended up to the fallback mechanism you see. 

P.s. I will remove the option from `CMakeLists` and it should be ok.